### ISSUE TITLE
Error when opening csv files, header row missing

### ISSUE THREAD

**Issue by mnebuerquo** [chrisbra/csv.vim#65]
I get the following error when opening a csv file:
```
"Locations.csv" [dos] 11L, 184C
Error detected while processing function CSVArrangeCol..<SNR>69_ArrangeCol..<SNR>69_Columnize:
line   99:
E767: Too many arguments to printf()
Error detected while processing function CSVArrangeCol:
line    2:
E171: Missing :endif
```

After that, it appears to work except the header row is missing.

Input file:
```
route_name,stop_sequence_number,stop_location_id,stop_action,stop_arrival_time,stop_departure_time,time_zone
MR-2001,1,9,Pickup,7/19/2014 8:00,7/19/2014 8:00,CST
MR-2001,2,6,Pickup,7/19/2014 10:00,7/19/2014 11:00,CST
MR-2001,3,9,Pickup,7/19/2014 13:00,7/19/2014 14:00,CST
MR-2001,4,8,Pickup,7/19/2014 16:00,7/19/2014 18:00,CST
MR-2001,5,4,Delivery,7/20/2014 8:00,7/20/2014 10:00,CST
MR-3005,1,8,Pickup,7/20/2014 8:00,7/20/2014 8:30,EST
MR-3005,2,2,Pickup,7/20/2014 10:00,7/20/2014 11:00,EST
MR-3005,3,7,Pickup,7/20/2014 13:00,7/20/2014 14:00,EST
MR-3005,4,8,Delivery,7/20/2014 23:00,7/27/2014 0:00,EST
MR-3005,5,6,Delivery,7/27/2014 8:00,7/27/2014 10:00,EST
MR-4006,1,7,Pickup,7/19/2014 8:00,7/19/2014 8:00,PST
MR-4006,2,2,Pickup,7/19/2014 10:00,7/19/2014 11:00,PST
MR-4006,3,9,Pickup,7/19/2014 13:00,7/19/2014 14:00,PST
MR-4006,4,10,Delivery,7/19/2014 16:00,7/19/2014 18:00,PST
MR-4006,5,2,Delivery,7/20/2014 8:00,7/20/2014 10:00,PST
MR-6509,1,4,Pickup,7/20/2014 8:00,7/20/2014 8:30,CST
MR-6509,2,10,Pickup,7/20/2014 10:00,7/20/2014 11:00,CST
MR-6509,3,1,Pickup,7/20/2014 13:00,7/20/2014 14:00,CST
MR-6509,4,3,Delivery,7/20/2014 23:00,7/27/2014 0:00,CST
MR-6509,5,5,Delivery,7/27/2014 8:00,7/27/2014 10:00,CST
MR-7250,1,4,Pickup,7/22/2014 8:00,7/22/2014 8:30,EST
MR-7250,2,3,Pickup,7/22/2014 10:00,7/22/2014 11:00,EST
MR-7250,3,6,Pickup,7/22/2014 13:00,7/22/2014 14:00,EST
MR-7250,4,6,Delivery,7/23/2014 17:00,7/23/2014 18:00,EST
MR-8000,1,7,Pickup,7/20/2014 8:00,7/20/2014 8:30,PST
MR-8000,2,2,Delivery,7/20/2014 10:00,7/20/2014 11:00,PST
MR-8000,3,3,Delivery,7/20/2014 15:00,7/20/2014 16:00,PST
```

**Comment by chrisbra** [comment]
What Vim version is this?

**Comment by mnebuerquo** [comment]
Sorry, forgot to include that in the initial report.
```
:version
VIM - Vi IMproved 7.4 (2013 Aug 10, compiled Jan  2 2014 19:39:32)
Included patches: 1-52
Modified by pkg-vim-maintainers@lists.alioth.debian.org
Compiled by buildd@
Huge version without GUI.  Features included (+) or not (-):
+acl             +conceal         +farsi           +libcall         +mouse_netterm   +profile         +syntax          +visualextra
+arabic          +cryptv          +file_in_path    +linebreak       +mouse_sgr       +python          +tag_binary      +viminfo
+autocmd         +cscope          +find_in_path    +lispindent      -mouse_sysmouse  -python3         +tag_old_static  +vreplace
-balloon_eval    +cursorbind      +float           +listcmds        +mouse_urxvt     +quickfix        -tag_any_white   +wildignore
-browse          +cursorshape     +folding         +localmap        +mouse_xterm     +reltime         -tcl             +wildmenu
++builtin_terms  +dialog_con      -footer          -lua             +multi_byte      +rightleft       +terminfo        +windows
+byte_offset     +diff            +fork()          +menu            +multi_lang      -ruby            +termresponse    +writebackup
+cindent         +digraphs        +gettext         +mksession       -mzscheme        +scrollbind      +textobjects     -X11
-clientserver    -dnd             -hangul_input    +modify_fname    +netbeans_intg   +signs           +title           -xfontset
-clipboard       -ebcdic          +iconv           +mouse           +path_extra      +smartindent     -toolbar         -xim
+cmdline_compl   +emacs_tags      +insert_expand   -mouseshape      -perl            -sniff           +user_commands   -xsmp
+cmdline_hist    +eval            +jumplist        +mouse_dec       +persistent_undo +startuptime     +vertsplit       -xterm_clipboard
+cmdline_info    +ex_extra        +keymap          +mouse_gpm       +postscript      +statusline      +virtualedit     -xterm_save
+comments        +extra_search    +langmap         -mouse_jsbterm   +printer         -sun_workshop    +visual          -xpm
   system vimrc file: "$VIM/vimrc"
     user vimrc file: "$HOME/.vimrc"
 2nd user vimrc file: "~/.vim/vimrc"
      user exrc file: "$HOME/.exrc"
  fall-back for $VIM: "/usr/share/vim"
Compilation: gcc -c -I. -Iproto -DHAVE_CONFIG_H     -g -O2 -fstack-protector --param=ssp-buffer-size=4 -Wformat -Werror=format-security -U_F
ORTIFY_SOURCE -D_FORTIFY_SOURCE=1
Linking: gcc   -Wl,-Bsymbolic-functions -Wl,-z,relro -Wl,--as-needed -o vim        -lm -ltinfo -lnsl  -lselinux  -lacl -lattr -lgpm -ldl
-L/usr/lib/python2.7/config-x86_64-linux-gnu -lpython2.7 -lpthread -ldl -lutil -lm -Xlinker -export-dynamic -Wl,-O1 -Wl,-Bsymbolic-functions
```

**Comment by chrisbra** [comment]
Hm, I can't reproduce this with Vim 7.4.52. Could it be, you have an older Vim installed as well and used that one? I suspect the %S identifier might be unknown to that Vim.

**Comment by jungle-boogie** [comment]
> Hm, I can't reproduce this with Vim 7.4.52.

Also not able to reproduce with 7.4.778 with the above data.

**Comment by mnebuerquo** [comment]
I see the same error in 7.3
```
:version
VIM - Vi IMproved 7.3 (2010 Aug 15, compiled May  4 2012 04:10:13)
Included patches: 1-429
Modified by pkg-vim-maintainers@lists.alioth.debian.org
Compiled by buildd@
Huge version with GTK2 GUI.  Features included (+) or not (-):
+arabic +autocmd +balloon_eval +browse ++builtin_terms +byte_offset +cindent +clientserver +clipboard +cmdline_compl +cmdline_hist
+cmdline_info +comments +conceal +cryptv +cscope +cursorbind +cursorshape +dialog_con_gui +diff +digraphs +dnd -ebcdic +emacs_tags +eval
+ex_extra +extra_search +farsi +file_in_path +find_in_path +float +folding -footer +fork() +gettext -hangul_input +iconv +insert_expand
+jumplist +keymap +langmap +libcall +linebreak +lispindent +listcmds +localmap +lua +menu +mksession +modify_fname +mouse +mouseshape
+mouse_dec +mouse_gpm -mouse_jsbterm +mouse_netterm -mouse_sysmouse +mouse_xterm +mouse_urxvt +multi_byte +multi_lang -mzscheme
+netbeans_intg +path_extra +perl +persistent_undo +postscript +printer +profile +python -python3 +quickfix +reltime +rightleft +ruby
+scrollbind +signs +smartindent -sniff +startuptime +statusline -sun_workshop +syntax +tag_binary +tag_old_static -tag_any_white +tcl
+terminfo +termresponse +textobjects +title +toolbar +user_commands +vertsplit +virtualedit +visual +visualextra +viminfo +vreplace
+wildignore +wildmenu +windows +writebackup +X11 -xfontset +xim +xsmp_interact +xterm_clipboard -xterm_save
   system vimrc file: "$VIM/vimrc"
     user vimrc file: "$HOME/.vimrc"
      user exrc file: "$HOME/.exrc"
  system gvimrc file: "$VIM/gvimrc"
    user gvimrc file: "$HOME/.gvimrc"
    system menu file: "$VIMRUNTIME/menu.vim"
  fall-back for $VIM: "/usr/share/vim"
Compilation: gcc -c -I. -Iproto -DHAVE_CONFIG_H -DFEAT_GUI_GTK  -pthread -I/usr/include/gtk-2.0 -I/usr/lib/i386-linux-gnu/gtk-2.0/include -I
/usr/include/atk-1.0 -I/usr/include/cairo -I/usr/include/gdk-pixbuf-2.0 -I/usr/include/pango-1.0 -I/usr/include/gio-unix-2.0/ -I/usr/include
/glib-2.0 -I/usr/lib/i386-linux-gnu/glib-2.0/include -I/usr/include/pixman-1 -I/usr/include/freetype2 -I/usr/include/libpng12     -g -O2 -fs
tack-protector --param=ssp-buffer-size=4 -Wformat -Wformat-security -Werror=format-security -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=1     -I/usr
/include/tcl8.5  -D_REENTRANT=1  -D_THREAD_SAFE=1  -D_LARGEFILE64_SOURCE=1
Linking: gcc   -L. -Wl,-Bsymbolic-functions -Wl,-z,relro -rdynamic -Wl,-export-dynamic  -Wl,-E  -Wl,-Bsymbolic-functions -Wl,-z,relro -Wl,--
as-needed -o vim   -lgtk-x11-2.0 -lgdk-x11-2.0 -latk-1.0 -lgio-2.0 -lpangoft2-1.0 -lpangocairo-1.0 -lgdk_pixbuf-2.0 -lcairo -lpango-1.0 -lfr
eetype -lfontconfig -lgobject-2.0 -lglib-2.0   -lSM -lICE -lXpm -lXt -lX11 -lXdmcp -lSM -lICE -lm -ltinfo -lnsl  -lselinux  -lacl -lattr -lg
pm -ldl  -L/usr/lib -llua5.1  -Wl,-E  -fstack-protector -L/usr/local/lib  -L/usr/lib/perl/5.14/CORE -lperl -ldl -lm -lpthread -lcrypt -L/usr
/lib/python2.7/config -lpython2.7 -lpthread -ldl -lutil -lm -Xlinker -export-dynamic -Wl,-O1 -Wl,-Bsymbolic-functions  -L/usr/lib -ltcl8.5 -
ldl -lpthread -lieee -lm -lruby1.8 -lpthread -lrt -ldl -lcrypt -lm  -L/usr/lib
```

Here is the vimrc I'm using for both machines, from my dotfiles in github:
https://github.com/mnebuerquo/dotfiles/blob/master/vim/vimrc

Here is a smaller file which reproduces the error:
```
id,location_name,state
1,plant 10,OH
2,plant 11,KY
3,plant 12,IN
4,plant 13,TN
5,plant 8,CA
6,plant 9,FL
7,supplier 1,AK
8,supplier 11,TX
9,supplier 12,GA
10,supplier 16,MA
```

**Comment by chrisbra** [comment]
Vim 7.3.429 was released more than 3 years ago and Vim version 7.3 is not supported anymore. If you can't upgrade your vim, please consider using an older version of the plugin available from the [vim.org](http://www.vim.org) page (e.g. [version 0.30](http://www.vim.org/scripts/download_script.php?src_id=21684) should still work)
